x-pack/plugins/observability_solution/synthetics/server/synthetics_service/project_monitor/project_monitor_formatter.ts

```diff
@@ -371,29 +383,30 @@ export class ProjectMonitorFormatter {
         monitors.map((m) => m.previousMonitor)
       );
 
-      const monitorsToUpdate = [];
-
-      for (let i = 0; i < decryptedPreviousMonitors.length; i++) {
-        const decryptedPreviousMonitor = decryptedPreviousMonitors[i];
-        const previousMonitor = monitors[i].previousMonitor;
-        const normalizedMonitor = monitors[i].monitor;
-
-        const {
-          attributes: { [ConfigKey.REVISION]: _, ...normalizedPreviousMonitorAttributes },
-        } = normalizeSecrets(decryptedPreviousMonitor);
-
-        const monitorWithRevision = formatSecrets({
-          ...normalizedPreviousMonitorAttributes,
-          ...normalizedMonitor,
-          revision: (previousMonitor.attributes[ConfigKey.REVISION] || 0) + 1,
-        });
-        monitorsToUpdate.push({
-          normalizedMonitor,
-          previousMonitor,
-          monitorWithRevision,
-          decryptedPreviousMonitor,
-        });
-      }
+      const monitorsToUpdate: MonitorConfigUpdate[] = [];
+
+      decryptedPreviousMonitors.forEach((decryptedPreviousMonitor) => {
+        const monitor = monitors.find((m) => m.previousMonitor.id === decryptedPreviousMonitor.id);
+        if (monitor) {
```

Comment:
monitor should always be found. If it's not, something is borked. Perhaps we should throw an error for !monitor

Reply:
i think since this is called with monitors that are first fetched and pushed via project monitor. i think error handling would be unecessary.